analytics-webapps/src/main/webapp/vue-app/generic-portlet/settings/tabs/ColorsSettingForm.vue

```diff
@@ -61,8 +61,7 @@ export default {
         || this.DEFAULT_COLORS;
     },
     setColor(color, index) {
-      this.colors.splice(index, 1, color);
-      this.settings.colors = this.colors;
+      this.settings.colors = this.colors.splice(index, 1, color);
```

Comment:
The splice method will apply a modification on the used array itself by deleting elements from `startIndex`.
https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Objets_globaux/Array/splice

The returned object of splice isn't the modified array but the deleted elements.

```javascript
a = ['1', '2', '3', '4'];
const indexOfElementToChange = 2;
const numberOfElementsToDelete = 3;
b = a.splice(indexOfElementToChange, numberOfElementsToDelete, '5');
console.log(a); // ['1', '2', '5']
console.log(b); // ['3', '4']
```

Reply:
aah ok !!Thanks, @boubaker  for the explanation